api/redhatopenshift/HcpCluster/hcpCluster-models.tsp

```diff
@@ -474,6 +478,18 @@ model NodePoolSpec {
   tuningConfigs?: string[];
 }
 
+/** Taint is controlling the node taint and its effects */
+model Taint {
+  /** The key of the taint */
+  key: string;
+
+  /** The value of the taint */
+  value?: string;
+
+  /** The effect of the taint */
+  effect: string;
```

Comment:
The only allowed "effect" values are: NoSchedule, PreferNoSchedule and NoExecute

We should at least document this

Reply:
I made it enum like union, so it is really clear for the user what can be assigned here.